6.x regression - Referencing `ext::auth::ClientTokenIdentity` from a link default ISEs

```
CREATE EXTENSION pgcrypto VERSION '1.3';
CREATE EXTENSION auth VERSION '1.0';
CREATE TYPE default::User;

ALTER TYPE default::User {
    CREATE REQUIRED LINK identity: ext::auth::Identity {
        SET default := (GLOBAL ext::auth::ClientTokenIdentity)
    };
};
```
produces
```
        File "/home/sully/src/e/edgedb/edb/schema/delta.py", line 999, in apply_subcommands
          schema = op.apply(schema, context=context)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/sully/src/e/edgedb/edb/schema/delta.py", line 3224, in apply
          schema = self._create_begin(schema, context)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/sully/src/e/edgedb/edb/pgsql/delta.py", line 5175, in _create_begin
          self._create_link(link, schema, orig_schema, context)
        File "/home/sully/src/e/edgedb/edb/pgsql/delta.py", line 5095, in _create_link
          self._alter_pointer_optionality(
        File "/home/sully/src/e/edgedb/edb/pgsql/delta.py", line 4275, in _alter_pointer_optionality
          (conv_expr_ctes, _, _) = self._compile_conversion_expr(
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/sully/src/e/edgedb/edb/pgsql/delta.py", line 4689, in _compile_conversion_expr
          ref_tables = schemamech.get_ref_storage_info(ir.schema, refs)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/sully/src/e/edgedb/edb/pgsql/schemamech.py", line 790, in get_ref_storage_info
          ptr = t.getptr(schema, s_name.UnqualName('id'))
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/sully/src/e/edgedb/edb/schema/sources.py", line 144, in getptr
          raise AssertionError(
      AssertionError: object type 'std::FreeObject' has no link or property 'id'
```

This is a reduced test case, but referencing it indirectly via another global fails also.

There are three components to this regression:
1. `id` was removed from `FreeObject` in 6.x
2. The definition of `ClientTokenIdentity` uses free objects to try to force materialization. This doesn't work anymore because of point 1, but I think also is not actually *necessary* right now. I'll follow up.
3. `_compile_conversion_expr`, used when a default is added, calls `get_ref_storage_info` on all the `Set`s in the `default`, and `get_ref_storage_info` assumes all objects have an `id`

It should be easy to fix point 3.
Point 2 is also worth fixing, but more annoying to backport to 6.x.

We can also fail without using `auth` with the quite contrived:
```
create type X;
alter type X { create required property y: str { set default := (with x := { y := "lol" }, select x.y) } };
```

--- a/edb/pgsql/schemamech.py
+++ b/edb/pgsql/schemamech.py
@@ -785,6 +785,8 @@ def get_ref_storage_info(
             source_typeref = ref.typeref
             if not irtyputils.is_object(source_typeref):
                 continue
+            if irtyputils.is_free_object(source_typeref):
+                continue
             schema, t = irtyputils.ir_typeref_to_type(schema, ref.typeref)
             assert isinstance(t, s_sources.Source)
             ptr = t.getptr(schema, s_name.UnqualName('id'))

--- a/tests/test_edgeql_ddl.py
+++ b/tests/test_edgeql_ddl.py
@@ -2046,6 +2046,28 @@ class TestEdgeQLDDL(tb.DDLTestCase):
             insert T;
         """)
 
+    async def test_edgeql_ddl_default_16(self):
+        await self.con.execute(r"""
+            create type T;
+            insert T;
+            alter type T {
+                create required property y: str {
+                    set default := (with x := { y := "lol" }, select x.y)
+                }
+            };
+            insert T;
+        """)
+
+        await self.assert_query_result(
+            r"""
+                select T { y }
+            """,
+            [
+                {'y': "lol"},
+                {'y': "lol"},
+            ],
+        )
+
     async def test_edgeql_ddl_default_circular(self):
         await self.con.execute(r"""
             CREATE TYPE TestDefaultCircular {

--- a/tests/test_http_ext_auth.py
+++ b/tests/test_http_ext_auth.py
@@ -347,6 +347,15 @@ class TestHttpExtAuth(tb.ExtAuthTestCase):
 
         CONFIGURE CURRENT DATABASE
         INSERT ext::auth::MagicLinkProviderConfig {{}};
+
+        # Pure testing code:
+        CREATE TYPE TestUser;
+        ALTER TYPE TestUser {{
+            CREATE REQUIRED LINK identity: ext::auth::Identity {{
+                SET default := (GLOBAL ext::auth::ClientTokenIdentity)
+            }};
+        }};
+
         """,
     ]
 

